File: app/src/main/java/com/stevenfrew/beatprompter/midi/Alias.java
package com.stevenfrew.beatprompter.midi;

import com.stevenfrew.beatprompter.R;
import com.stevenfrew.beatprompter.SongList;

import java.util.ArrayList;
import java.util.List;

public class Alias {
    public String mName;
    private List<AliasComponent> mComponents;

    public Alias(String name)
    {
        mName=name;
        mComponents=new ArrayList<>();
    }
    public Alias(String name,List<AliasComponent> components)
    {
        this(name);
        mComponents=components;
    }

    public List<OutgoingMessage> resolve(List<Alias> aliases,byte[] arguments, byte channel) throws ResolutionException
    {
        ArrayList<OutgoingMessage> outMessages=new ArrayList<>();
        for(AliasComponent component:mComponents)
            outMessages.addAll(component.resolve(aliases,arguments,channel));
        return outMessages;
    }
}

File: app/src/main/java/com/stevenfrew/beatprompter/midi/Alias.kt
package com.stevenfrew.beatprompter.midi

class Alias constructor(name:String,components:List<AliasComponent>)
{
    @JvmField val mName=name;
    private val mComponents=components;

    @Throws(ResolutionException::class)
    fun resolve(aliases: List<Alias>, arguments: ByteArray, channel: Byte): List<OutgoingMessage> {
        return mComponents.flatMap{it.resolve(aliases, arguments, channel)}
    }
}